### DISCUSSION TITLE
Too many Open files

### DISCUSSION THREAD

**Discussion by kpot86** [relayrides/pushy#925] [Q&A]
I have an application that uses maven dependency for pushy 0.14 version. 
Ever since we included it in the project we see an error "Too many open files" that causes WidFly to crash. 
Do you have any idea were we could look to solve this.

**Comment by jchambers** [comment]
The main thing Pushy does that consumes file descriptors is opening connections to APNs servers. Do you have a large number of `ApnsClient` instances at any one time? Have you set a high concurrent connection count?
